FreeRTOS-Plus/Source/Application-Protocols/network_transport/freertos_plus_tcp/using_plaintext/using_plaintext.c

```diff
@@ -131,6 +131,14 @@ int32_t Plaintext_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
                                   bytesToRecv,
```

Comment:
We also want to prevent FreeRTOS_recv() from blocking when it is called to see if there is any data available, rather than when it is called because it is know there is a packet being received.  If the number of bytes requested is 1, then it is being called to look for the start of another frame, so we can prevent it blocking when no frame is available by adding:

```
    if( ( bytesToRecv > 1 ) || ( FreeRTOS_recvcount( pNetworkContext->tcpSocket ) > 0 ) )
    {
        socketStatus = FreeRTOS_recv( pNetworkContext->tcpSocket, pBuffer, bytesToRecv, 0 );
    }
    else
    {
        socketStatus = 0;
    }
```

Reply:
That's a nice addition. One thing I noticed is that `FreeRTOS_recvcount` may return `-pdFREERTOS_ERRNO_EINVAL` for an invalid socket, and we don't necessarily want that to be retriable and return 0. I've added a check for this edge case.